File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board):
        seen = set()
        
        for i in range(9):
            for j in range(9):
                number = board[i][j]
                if number != '.':
                    row = "{} seen in row {}".format(number, i)
                    col = "{} seen in col {}".format(number, j)
                    box = "{} seen in box {}-{}".format(number, i // 3, j // 3)

                    if (row in seen) or (col in seen) or (box in seen):
                        return False

                    seen.add(row)
                    seen.add(col)
                    seen.add(box)

        return True

File: 0036-valid-sudoku/0036-valid-sudoku.cpp
#include <vector>
#include <set>
#include <string>

class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        std::set<std::string> seen;
        
        for (int i = 0; i < 9; ++i) {
            for (int j = 0; j < 9; ++j) {
                char number = board[i][j];
                if (number != '.') {
                    std::string row = std::to_string(number) + " seen in row " + std::to_string(i);
                    std::string col = std::to_string(number) + " seen in col " + std::to_string(j);
                    std::string box = std::to_string(number) + " seen in box " + std::to_string(i / 3) + "-" + std::to_string(j / 3);

                    if (seen.find(row) != seen.end() || seen.find(col) != seen.end() || seen.find(box) != seen.end()) {
                        return false;
                    }
                    
                    seen.insert(row);
                    seen.insert(col);
                    seen.insert(box);
                }
            }
        }
        
        return true;
    }
};
